### PR TITLE
Modified code to incorporate changes made in NDO version >= 4.2's API response for 'regions'

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

**Review by akinross — DISMISSED** [review]
LGTM

File: mso/resource_mso_schema_site_vrf_region.go
Comment (sajagana) [inline mso/resource_mso_schema_site_vrf_region.go:1]: `hub_network` - Causing the conflicts during the reapply(idempotent).

**Review by sajagana — APPROVED** [review]
The current change looks good! but resource required additional fix to handle idempotent check.

**Review by akinross — APPROVED** [review]

**Review by sajagana — APPROVED** [review]
LGTM!

File: mso/resource_mso_schema_site_vrf_region.go [hunk 1]
```diff
@@ -519,7 +530,7 @@ func resourceMSOSchemaSiteVrfRegionRead(d *schema.ResourceData, m interface{}) e
 						}
 						apiRegion := models.StripQuotes(regionCont.S("name").String())
 						if apiRegion == stateRegion {
-							d.SetId(apiRegion)
+							d.SetId(fmt.Sprintf("%s/sites/%s/template/%s/vrf/%s/region/%s", schemaId, stateSite, stateTemplate, stateVrf, stateRegion))
```
Comment (akinross) [inline mso/resource_mso_schema_site_vrf_region.go:533]: Should this site id not be set to something as below: 

`fmt.Sprintf("%s/sites/%s-%s/anps/%s/epgs/%s/domainAssociations/%s", schemaId, apiSite, apiTemplate, apiAnp, apiEPG, apiDomain)`

File: mso/resource_mso_schema_site_vrf_region.go
Comment (shrsr) [reply mso/resource_mso_schema_site_vrf_region.go:1]: Took care of these

File: mso/resource_mso_schema_site_vrf_region.go (hunk 1 above)
Comment (shrsr) [reply mso/resource_mso_schema_site_vrf_region.go:533]: I set the ID in a format which is equivalent to the ID we provide for import?

Comment (shrsr) [reply mso/resource_mso_schema_site_vrf_region.go:533]: We need to discuss this and finalize a format for all the resources as it's done differently in different resources.
I'm resolving this for now.

**Review by akinross — APPROVED** [review]

**Review by anvitha-jain — APPROVED** [review]

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**